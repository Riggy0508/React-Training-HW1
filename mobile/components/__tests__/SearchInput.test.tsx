import React from 'react';
import {render, fireEvent} from '@testing-library/react-native'
import Search from '../SearchInput'

jest.mock('expo-font');


describe('Search',()=>{
    it('display value in input field',()=>{
        const{getByLabelText}=render(
            <Search value='sample text' onSearch={()=>{}}/>,
        )
        const input=getByLabelText('Text input field');
        expect(input.props.value).toBe('sample text');
    })

    it('render the search component',()=>{
        const { getByLabelText } = render(<Search value="" onSearch={() => {}} />);
        const input = getByLabelText('Text input field');
        expect(input).toBeTruthy();
  })
})
