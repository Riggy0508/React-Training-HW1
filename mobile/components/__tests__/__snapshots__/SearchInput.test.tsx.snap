// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default version of the FAB component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#CCCCCC",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 24,
      "paddingHorizontal": 23,
      "width": 300,
    }
  }
>
  <TextInput
    accessibilityLabel="Text input field"
    onChangeText={[MockFunction]}
    placeholder="Filter lotteries"
    style={
      {
        "flex": 1,
        "fontSize": 16,
        "height": 40,
        "paddingRight": 10,
      }
    }
    value=""
  />
  <Text />
</View>
`;

exports[`Search Render default version of the FAB component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#CCCCCC",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 24,
      "paddingHorizontal": 23,
      "width": 300,
    }
  }
>
  <TextInput
    accessibilityLabel="Text input field"
    onChangeText={[MockFunction]}
    placeholder="Filter lotteries"
    style={
      {
        "flex": 1,
        "fontSize": 16,
        "height": 40,
        "paddingRight": 10,
      }
    }
    value=""
  />
</View>
`;
